routes/actions.js

```diff
@@ -210,7 +219,16 @@ router.get('/register', function(req, res) {
 });
 
 router.get('/register/:code', function(req, res, next) {
-  const { code } = req.params;
+  const { code } = req.params.code;
+  if (req.query.signupLanguage && languages.isValid(req.query.signupLanguage)){
+    const lang = req.query.signupLanguage;
+    let maxAge = 1000 * 60 * config.sessionCookieDuration; // cookie age: 30 days
+    res.cookie('locale', lang, {
+        maxAge,
+        httpOnly: true
+        });
+    i18n.setLocale(req, lang);
+  }
   InviteLink
```

Comment:
Per DRY (don't repeat yourself) principle, I would suggest removing this into its own dedicated function so we don't have an exact copy of the same code in two places in the same file.